packages/@angular/cli/commands/config.ts

```diff
@@ -0,0 +1,135 @@
+import { Command } from '../models/command';
+import * as fs from 'fs';
+import { CliConfig } from '../models/config';
+import { oneLine } from 'common-tags';
+
+const SilentError = require('silent-error');
+
+
+export interface ConfigOptions {
+  jsonPath: string;
+  value?: string;
+  global?: boolean;
+}
+
+export default class ConfigCommand extends Command {
+  public readonly name = 'config';
+  public readonly description = 'Get/set configuration values.';
+  public readonly arguments = ['jsonPath', 'value'];
+  public readonly options = [
+    {
+      name: 'global',
+      type: Boolean,
+      'default': false,
+      aliases: ['g'],
+      description: 'Get/set the value in the global configuration (in your home directory).'
+    }
+  ];
+
+  public run(options: ConfigOptions) {
+    const config = options.global ? CliConfig.fromGlobal() : CliConfig.fromProject();
+
+    if (config === null) {
+      throw new SilentError('No config found. If you want to use global configuration, '
```

Comment:
Use `logger.error` and return an errorCode (or just return)? does the refactoring support returning a number?  It'd be nice if we could remove the need to use `SilentError` and only deal with exceptions propagating up from here when something unexpected or fatal occurs.

Reply:
The basis is that failing command runs throw an exception, but what should happen here is that this check should be run during the validate method.

Reply:
I suppose that basis is what I'm questioning which is really more systemic than particular to this command.  There's failing commands due to unexpected behavior (bad logic, memory issues, etc.)  and failing commands due to do expected behavior (in this case the config not found or improper value).  The first should definitely give rise to an exception but the second seems like something that could be better handled locally.  Control flow via exception can be highly problematic.  A try/catch in the wrong place and the user never knows something bad happened.  An error is also not necessarily fatal.  So potentially, using the `logger.error` for expected user errors and reserving `logger.fatal` for actual exceptions caught at a higher level could help the user understand the severity of the situation.  As opposed to the current version, where error reporting is fairly inconsistent and confusing.